app/src/main/java/com/example/wellfed/recipe/RecipeDB.java

```diff
@@ -69,29 +91,37 @@ public void addRecipe(Recipe recipe) throws InterruptedException {
 
             return null;
         })
-                .addOnSuccessListener(unused -> {
-                    Log.d(TAG, "onSuccess: ");
-                    addLatch.countDown();
-                })
-                .addOnFailureListener(e -> {
-                    Log.d(TAG, "onFailure: ");
-                    addLatch.countDown();
-                });
+        .addOnSuccessListener(unused -> {
+            Log.d(TAG, "onSuccess: ");
+            addLatch.countDown();
+        })
+        .addOnFailureListener(e -> {
+            Log.d(TAG, "onFailure: ");
+            addLatch.countDown();
+        });
 
         addLatch.await();
```

Comment:
Just a thought, this will block our UI while executing. I am not sure but I will keep this in mind to read in future if it poses a problem